Retention at Rank I: Gui (#142)
polkadot-fellows/Evaluations: evidence/gui/0002-rank-1-retention.md

# Argument-0002: Retention at Rank I

|                 |                                                                                             |
| --------------- | ------------------------------------------------------------------------------------------- |
| **Report Date** | Date of submission (2025/04/05)                                                             |
| **Submitted by**| Guillaume Thiolliere                                                                        |

## Member details

- Matrix username: @guillaume:parity.io
- Polkadot address: 13psJuWEjBuZGaFqXvFnLMC6ME8RVVfQAhtFhydYjW45oKgZ
- Current rank: I
- Date of initial induction: 2025/01/09
- Date of last report: 2025/01/09
- Area(s) of Expertise/Interest: 
  - polkadot business-logic (aka the 'runtime')
  - the internals of the frame pallet framework
  - runtime and host APIs

## Reporting period
- Start date: 2025/02/06
- End date: 2025/04/05

> ## Argument

Reviewer: Please make sure to provide sufficient details about your work for this past period (e.g. more information Proof of Personhood, transaction extension horizon, and contributions to the runtimes upgrade).
Your argument needs to be presented in enough depth to allow anybody who reads it to evaluate how well your work matches the expectations of Rank 1 members laid out in the Manifesto, such as:
![Rank1_Retentionexamples](https://github.com/user-attachments/assets/25aeb0a0-b83d-4891-bca2-8ef142c64819)

Author: I updated to make it more precise

Reviewer: Please get the argument spell-checked: https://www.grammarly.com/grammar-check

Reviewer: Thank you! 
Fellows will now review and share their feedback, if needed.

Author: Thank you :-)

# Argument-0002: Retention at Rank I

|                 |                                                                                             |
| --------------- | ------------------------------------------------------------------------------------------- |
| **Report Date** | Date of submission (2025/04/05)                                                             |
| **Submitted by**| Guillaume Thiolliere                                                                        |

## Member details

- Matrix username: @guillaume:parity.io
- Polkadot address: 13psJuWEjBuZGaFqXvFnLMC6ME8RVVfQAhtFhydYjW45oKgZ
- Current rank: I
- Date of initial induction: 2025/01/09
- Date of last report: 2025/01/09
- Area(s) of Expertise/Interest: 
  - polkadot business-logic (aka the 'runtime')
  - the internals of the frame pallet framework
  - runtime and host APIs

## Reporting period
- Start date: 2025/02/06
- End date: 2025/04/05

## Argument

> I focused mainly on proof of personhood with George and Michal. Part of this work is now [public](https://github.com/paritytech/polkadot-sdk/pull/8164).
> In this PR I implemented the origin restriction pallet, it is a pallet to limit usage of certain origins. This will be used to allow usage to recognized people. It is also used to limited allow some usage to some entities such as candidates while preventing spam.
> I implemented the transaction extension in the people pallet, those transaction extension allows to authentificate as a personal id or as an alias. I also implemented the replay protection in those transaction extensions.
> I reviewed work from George and Michal for the other part of the pallet.
> 
> Apart from proof of personhood, I reviewed several PRs in polkadot-sdk and submitted minor ones, and reviewed the upgrade of runtimes to version 2412.
> 
> I continued working on the transaction extension horizon. [A PR](https://github.com/paritytech/parity-scale-codec/pull/715) is currently open, aimed at simplifying the migration of unsigned transactions to general transactions, which is undergoing an audit. [Another PR](https://github.com/paritytech/polkadot-sdk/pull/6325) facilitates migrating existing unsigned transactions in polkadot-sdk to general transactions in a backward-compatible manner.

Author: ```suggestion
I focused mainly on proof of personhood with George and Michal. Part of this work is now [public](https://github.com/paritytech/polkadot-sdk/pull/8164).
In this PR I implemented the origin restriction pallet, it is a pallet to limit usage of certain origins. This will be used to restrict usage from recognized people and also candidates for some DIM.
I implemented the transaction extension in the people pallet, those transaction extension allows to authenticate as a personal id or as an alias. I also implemented the replay protection.
I reviewed work from George and Michal for the other part of the pallet.

Apart from proof of personhood, I reviewed several PRs in polkadot-sdk, submitted minor ones, and reviewed the upgrade of runtimes to version 2412.

I continued working on the transaction extension horizon. [A PR](https://github.com/paritytech/parity-scale-codec/pull/715) is currently open, aimed at simplifying the migration of unsigned transactions to general transactions, which is undergoing an audit. [Another PR](https://github.com/paritytech/polkadot-sdk/pull/6325) migrates some existing unsigned transactions in polkadot-sdk to general transactions in a backward-compatible manner.
```
